### PR TITLE
Refactor: Floorbot Overlay + Sprite Reduction

### PR DESCRIPTION
**Pull request by Oyu07**
## What Does This PR Do
Reduces amount of floorbot sprites from 49 to 26
Switched from sprites/icons to overlays

Naming format
[color]floorbot_[overlay names] 

Overlay Names
floorbot_+
- on | off
- metal
- work

## Why It's Good For The Game
Less sprites, aibots.dmi is already massive and has multiple unused and defunct sprites (will go over them at a later date confirming each)

## Testing
1. Spawn floorbot
2. On/Off updates overlay - Double confirm correct sprite/overlays are used
3. Var edit tile amount to 0, update bot on/off for update, check metal is gone
4. Spawn metal, allow it to replenish, metal overlay applies
5. Same but manually feed metal/tiles
6. Check working animation

## Changelog
NPFC

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 1]
```diff
@@ -389,13 +389,15 @@
...
 	if(mode == BOT_REPAIRING)
-		icon_state = "[toolbox_color]floorbot-c"
-		return
-	if(amount > 0)
-		icon_state = "[toolbox_color]floorbot[on]"
+		. += "[toolbox_color]floorbot_work]"
 	else
-		icon_state = "[toolbox_color]floorbot[on]e"
+		. += "[toolbox_color]floorbot_[on ? "on" : "off"]"
+		. += "[toolbox_color]floorbot_[amount > 0 ? "metal" : "" ]"
```
[the remark this answers is not in the document]
Comment (Oyu07) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:400]: and this is why I request reviews :)
Thx for catching

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 2]
```diff
@@ -389,13 +389,15 @@
...
+	return
+
+/mob/living/simple_animal/bot/floorbot/update_overlays()
+	. = ..()
 	if(mode == BOT_REPAIRING)
-		icon_state = "[toolbox_color]floorbot-c"
-		return
-	if(amount > 0)
-		icon_state = "[toolbox_color]floorbot[on]"
+		. += "floorbot_work]"
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:397]:
```suggestion
		. += "floorbot_work"
```

Comment (Oyu07) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm]: [f8c0537](https://github.com/ParadiseSS13/Paradise/pull/23081/commits/f8c053744eb7ece0542400cd10625fd6edd13fdc)

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 3]
```diff
@@ -389,13 +389,15 @@
...
 	if(mode == BOT_REPAIRING)
-		icon_state = "[toolbox_color]floorbot-c"
-		return
-	if(amount > 0)
-		icon_state = "[toolbox_color]floorbot[on]"
+		. += "floorbot_work"
 	else
-		icon_state = "[toolbox_color]floorbot[on]e"
+		. += "floorbot_[on ? "on" : "off" ]"
+		. += "floorbot_[amount > 0 ? "metal" : "" ]"
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:400]:
```suggestion
		. += "floorbot_[on ? "on" : "off"]"
		. += "floorbot_[amount > 0 ? "metal" : ""]"
```
I do think we use without spacing at the end more over the code.

**Review by Henri215 — APPROVED** [review]
Lewcs Good To Me after this last review is resolved

File: code/modules/mob/living/simple_animal/bot/floorbot.dm (hunk 3 above)
Comment (Oyu07) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm:400]: I will double check and return

Comment (Oyu07) [reply code/modules/mob/living/simple_animal/bot/floorbot.dm]: [5e2e844](https://github.com/ParadiseSS13/Paradise/pull/23081/commits/5e2e8449ed26be446b65b54245580a6eb123435b)

**Review by Sirryan2002 — APPROVED** [review]

**Review by S34NW — APPROVED** [review]
deconflict for merge

**Comment by Oyu07** [comment]
@S34NW Ready for merge

**Pull request merged**